app/components/announcement_banner/announcement_banner.tsx

```diff
@@ -0,0 +1,187 @@
+    allowDismissal: boolean;
+}
+
+const getStyle = makeStyleSheetFromTheme((theme: Theme) => ({
+    background: {
+        backgroundColor: theme.sidebarBg,
+    },
+    bannerContainer: {
+        flex: 1,
+        paddingHorizontal: 10,
+        overflow: 'hidden',
+        flexDirection: 'row',
+        alignItems: 'center',
+        marginHorizontal: 8,
+        borderRadius: 7,
+    },
+    wrapper: {
+        flexDirection: 'row',
+        flex: 1,
+        overflow: 'hidden',
+    },
+    bannerTextContainer: {
+        flex: 1,
+        flexGrow: 1,
+        marginRight: 5,
+        textAlign: 'center',
+    },
+    bannerText: {
+        ...typography('Body', 100, 'SemiBold'),
+    },
+}));
+
+const CLOSE_BUTTON_ID = 'announcement-close';
+
+const BUTTON_HEIGHT = 48; // From /app/utils/buttonStyles.ts, lg button
+const TITLE_HEIGHT = 30 + 12; // typography 600 line height
+const MARGINS = 12 + 24 + 10; // (after title + after text + after content) from ./expanded_announcement_banner.tsx
+const TEXT_CONTAINER_HEIGHT = 150;
+const DISMISS_BUTTON_HEIGHT = BUTTON_HEIGHT + 10; // Top margin from ./expanded_announcement_banner.tsx
+
+const SNAP_POINT_WITHOUT_DISMISS = TITLE_HEIGHT + BUTTON_HEIGHT + MARGINS + TEXT_CONTAINER_HEIGHT;
+
+const AnnouncementBanner = ({
+    bannerColor,
+    bannerDismissed,
+    bannerEnabled,
+    bannerText = '',
+    bannerTextColor = '#000',
+    allowDismissal,
+}: Props) => {
+    const intl = useIntl();
+    const serverUrl = useServerUrl();
+    const height = useSharedValue(0);
+    const theme = useTheme();
+    const [visible, setVisible] = useState(false);
+    const style = getStyle(theme);
+
+    const renderContent = useCallback(() => (
+        <ExpandedAnnouncementBanner
+            allowDismissal={allowDismissal}
+            bannerText={bannerText}
+        />
+    ), [allowDismissal, bannerText]);
+
+    const handlePress = useCallback(() => {
+        const title = intl.formatMessage({
+            id: 'mobile.announcement_banner.title',
+            defaultMessage: 'Announcement',
+        });
+
+        let snapPoint = SNAP_POINT_WITHOUT_DISMISS;
+        if (allowDismissal) {
+            snapPoint += DISMISS_BUTTON_HEIGHT;
+        }
+
+        bottomSheet({
+            closeButtonId: CLOSE_BUTTON_ID,
+            title,
+            renderContent,
+            snapPoints: [snapPoint, 10],
+            theme,
+        });
+    }, [theme.sidebarHeaderTextColor, intl.locale, renderContent, allowDismissal]);
+
+    const handleDismiss = useCallback(() => {
+        dismissAnnouncement(serverUrl, bannerText);
+    }, [serverUrl, bannerText]);
+
+    useEffect(() => {
+        const showBanner = bannerEnabled && !bannerDismissed && Boolean(bannerText);
+        setVisible(showBanner);
+    }, [bannerDismissed, bannerEnabled, bannerText]);
+
+    useEffect(() => {
+        height.value = withTiming(visible ? ANNOUNCEMENT_BAR_HEIGHT : 0, {
+            duration: 500,
```

Comment:
half a second seems too much, what about a standard 300ms ? or even using the default value of the `withTiming` function?

Reply:
Probably I left it like that to make sure I was able to see the animation. @matthewbirtch not sure if you have any special requirement on this detail. (The speed at which the banner shows / hides)

Reply:
Nice catch @enahum. I agree that 500ms is too slow. I would say 200ms would be better